core/vm.c

```diff
@@ -35,9 +35,9 @@
 #include "include/hax_core_interface.h"
 #include "include/ept.h"
 #include "include/paging.h"
+#include "include/config.h"
 
-static uint8_t vm_mid_bits = 0;
-#define VM_MID_BIT 8
+static uint64_t hax_max_vms_bits = 0;
```

Comment:
Since this variable is local to `vm.c`, we don't really have to rename it. If you don't like the old name, how about `vm_mid_seed`? It's based on the pattern of `vm_t::vpid_seed` (`core/include/vm.h`), and implies that it's used for generating VM IDs. Actually, we can follow `vpid_seed`'s example and turn this variable into an array, which should be a better way to make a connection between it and `HAX_MAX_VMS`:

```suggestion
static uint8_t vm_mid_seed[HAX_MAX_VMS / 8];
```

Note that we don't really need to initialize this array, since the default behavior is to fill it with zeros.